### DISCUSSION TITLE
Saleor as a marketplace

### DISCUSSION THREAD

**Discussion by looped-monk** [saleor/saleor#10641] [General] [answered]
I was checking out the dashboard and couldnt anything pertaining if one wants to run Saleor as a marketplace. 

Is there any plugin or feature available pertaining to the same ? 

@korycins @maarcingebala Any update on Saleor as a marketplace ?

**Comment by peelar** [comment]
Hello @narayanan-ka 👋 

Can you explain what you mean by "running Saleor as a marketplace"?

**Reply by looped-monk** [reply]
Hello. I intend to Run Saleor configurable through the App(since this feature is to avoid touching the core) AS a marketplace where each user can be a customer as well as a seller. I'm intending to build an ecommerce app connecting to saleor docker whereby each user can be

- be a customer and buy products from other users(vendors)
- a vendor(seller) where he/she can sell products
- My end as a marketplace host receives a cut for the same as a comission % of the sale value. 

Similar to how popular marketplaces like amazon work.

**Reply by looped-monk** [reply]
> Hello. I intend to Run Saleor configurable through the App(since this feature is to avoid touching the core) AS a marketplace where each user can be a customer as well as a seller. I'm intending to build an ecommerce app connecting to saleor docker whereby each user can be
> 
> - be a customer and buy products from other users(vendors)
> - a vendor(seller) where he/she can sell products
> - My end as a marketplace host receives a cut for the same as a comission % of the sale value. 
> 
> Similar to how popular marketplaces like amazon work.

@peelar hope I have sufficiently answered your question.

**Reply by looped-monk** [reply]
@korycins Could you help out ?

**Reply by looped-monk** [reply]
> > Hello. I intend to Run Saleor configurable through the App(since this feature is to avoid touching the core) AS a marketplace where each user can be a customer as well as a seller. I'm intending to build an ecommerce app connecting to saleor docker whereby each user can be
> > 
> > - be a customer and buy products from other users
> > - a seller where he/she can sell products
> > - My end as a marketplace host receives a cut for the same as a comission % of the sale value. 
> > 
> > Similar to how popular marketplaces like amazon work.
> 
> @peelar hope i have sufficiently answered your question.

@peelar Can i expect a reply after detailing out an answer based on your question.

**Reply by looped-monk** [reply]
@shovan777 Any inputs ?

**Reply by maarcingebala** [reply]
Hi @narayanan-ka, Saleor was never designed as a marketplace platform, so it won't work out of the box. The dashboard also doesn't support it, e.g. there is no way to allow certain users to log in to the dashboard and manage only selected subset of products.

**Reply by vitviggs** [reply]
> Hello @narayanan-ka 👋
> 
> Can you explain what you mean by "running Saleor as a marketplace"?

I'ts Multivendor

**Comment by leandrobortolotto** [comment]
Thank you for your answers. Very helpful and time-saving. Based on this, I must move to a different solution since I need different Partners (Vendors/Sellers).

**Reply by looped-monk** [reply]
@leandrobortolotto Check the link below. Exactly for your use case and mine.

https://github.com/django-oscar/

**Reply by leandrobortolotto** [reply]
Hi @narayanan-ka. Thank you for sharing. It's not clear to me the marketplace / multi-vendor availability. Did you find this feature?

Thank you.

**Reply by looped-monk** [reply]
https://stackoverflow.com/questions/48096920/django-oscar-and-multitenancy

Let me know if this helps. I'm reading it too. Had read earlier once and thought it's good for a start.

Besides there is Saloer multi-vendor too but the payment workload is incomplete.

**Reply by looped-monk** [reply]
Saleor Mulit vendor which is currently under development. I didn't fork it since its incomplete and under development. 


https://github.com/Saleor-Multi-Vendor/saleor-multi-vendor

**Reply by leandrobortolotto** [reply]
Thanks for sharing, @narayanan-ka. I'm looking for similar solutions as well. I'll develop a lot on top of the marketplace.

**Reply by looped-monk** [reply]
@leandrobortolotto which repo are you planning to start with since both are an option ?

**Reply by leandrobortolotto** [reply]
Honestly, I'm not sure, since I'm a Java developer and I don't have experience with Python or Ruby. But, I can always learn to develop it. In any case, I took a look at this one: https://github.com/shuup/ (that is in the StackOverflow link you shared) which seems to have a marketplace/multi-vendor implemented.

** Although this comment is very much not encouraging: https://github.com/shuup/shuup/issues/2626 **

I like the Saleor project itself, its architecture, documentation, etc. But I do need something more ready for production. My deadline is very short. Maybe this Shuup can be a solution.

What is your opinion on that one?

**Reply by looped-monk** [reply]
Both saleor and Oscar don't have multi vendor marketplace capabilities as a default capability. What I loved about saleor is not touching the core and the implementation of  graphql query language) alongwith a great dashboard interface. Rest core features remain the same more or less.

However ,  transforming it to a marketplace/mulivendor requires a lot of modifications. The saleor multi vendor has approached it I beleive by touching the core and a critical payment workflow  component is pending. Also if one needs to setup External user synch , which I have impmemented for the time being but I'm short of transforming it into a marketplace model. I'm now beginning to look at alternatives as im beginning to lose patience.

Django Oscar can be implemented right inside your core django project(if that's the case), and modifications done to the codebase so one doesn't need the users to leave the database for authentication.

Time & skill wise , Saleor requires more work since the additonal Graphql has to be learnt by most developers. Django oscar is straightforward implemention and modifications will take relatively far lesser time.

Im currently looking at Shuup. However the security aspect and lack of updates to the repo is bothering me.

**Reply by leandrobortolotto** [reply]
Maybe we can program it together and share the code later. It can speed up the process for the two of us. Do you program in Java? Just an idea.

**Reply by looped-monk** [reply]
@leandrobortolotto Its nevertheless a good idea but im more inclined towards python and django.

**Reply by looped-monk** [reply]
@leandrobortolotto i found this which is useful 

https://django-oscar.readthedocs.io/en/latest/ref/apps/dashboard.html

**Reply by leandrobortolotto** [reply]
@narayanan-ka That looks great. I'm reading the documentation. If I understood it correctly, the Partner App would be the Multi-Vendor feature, right?

**Reply by leandrobortolotto** [reply]
So, Shuup is not an option either: https://shuup-guide.readthedocs.io/en/latest/multivendor.html - It says the Multivendor application is only available under a license.

**Reply by looped-monk** [reply]
@leandrobortolotto yup. Just read about that. It's a paid version. I'm currently looking to implement django oscar marketplace

**Reply by leandrobortolotto** [reply]
@narayanan-ka what do you mean by that? I have no reservations about any technology. The thing is, as long as it solves my problem and it has a strong architecture because I'm looking for something long-term. Did you have a good experience with django-oscar?

I need headless multivendor e-commerce.

**Reply by leandrobortolotto** [reply]
@narayanan-ka I found this option too: https://www.sharetribe.com/how-to-develop/open-source-marketplace/#sharetribe-go-community-edition

https://github.com/sharetribe

But, from all the options, I still like Saleor the best. Even if that means implementing the marketplace features.

**Reply by looped-monk** [reply]
@leandrobortolotto Touching this thread after quite a while. Any update on your end regarding a marketplace solution. Saleor -multi-vendor strangely has no furthur development since the last time we spoke.

**Reply by leandrobortolotto** [reply]
@narayanan-ka Choosing a multi-vendor marketplace software is on our agenda for the beginning of 2023. Saleor is on our list. But, the little I analyzed the code, I think it will be necessary to change the structure to adapt Saleor to be a multi-vendor from its core. Not sure if the https://github.com/Saleor-Multi-Vendor/saleor-multi-vendor plug-in will satisfy all the needs we have. Let's see.

**Reply by looped-monk** [reply]
Yup. Saleor core needs to get modified quite a lot from what I've read. However Saleor multi vendor has done some good amount of work but still some work is pending. I saw your reply there offering help to continue the pending tasks and remembered we had spoken too here. Nevertheless, It's on my agenda too for a marketplace solution on a priority basis.i just had a word with the Owners of saleor core and their plans for a multi vendor and its not on their agenda for 2023. So we are on our own. Lets touch base sooner.

**Reply by vitviggs** [reply]
> @narayanan-ka Choosing a multi-vendor marketplace software is on our agenda for the beginning of 2023. Saleor is on our list. But, the little I analyzed the code, I think it will be necessary to change the structure to adapt Saleor to be a multi-vendor from its core. Not sure if the https://github.com/Saleor-Multi-Vendor/saleor-multi-vendor plug-in will satisfy all the needs we have. Let's see.

looks like an abandoned project

**Comment by timuric** [comment, marked as answer]
You can use Saleor for marketplace, but you would need to create your own admin interface for your vendors (they're users that did that). There is no way for Saleor (or any other platform) to know rules of your platform or what kind of permissions your vendors have. For example, do vendors control shipping rate, taxes, inventory, listing etc. 

There are hundreds of choices to be made when creating a marketplace, similar to creating storefronts, there is simply no one size fits all.

**Reply by leandrobortolotto** [reply]
@timuric thank you very much for your answer. You're saying that we only must create the Frontend considering filters by Vendor, etc? Do you have an example of a marketplace (multi-vendor) running in production using Saleor? Can you please connect me to someone that did it? I would love to know more about it. Thank you.

**Reply by looped-monk** [reply]
Hi @leandrobortolotto Was discussing with @timuric on this thread about their plans for 2023. They are inviting suggestions and if feasible to implement, will do so. Suggest you take a look.
https://github.com/saleor/saleor/discussions/11608

**Reply by timuric** [reply]
@leandrobortolotto yes we have, but am not allowed to share since they are customers of Saleor cloud. 
What you are looking for is most likely a custom admin panel that would allow managing some of the configuration of the store. You can use https://github.com/saleor/saleor-dashboard as a reference for building your own vendor admin panel.

**Reply by SpencerPatterson** [reply]
Is there any plan for Saleor to ever go "Multi-Vendor" natively or do we need to need to just build custom?  I know I asked this about three years ago because it was on the roadmap and then it disappeared.

**Reply by leandrobortolotto** [reply]
> @leandrobortolotto yes we have, but am not allowed to share since they are customers of Saleor cloud. What you are looking for is most likely a custom admin panel that would allow managing some of the configuration of the store. You can use https://github.com/saleor/saleor-dashboard as a reference for building your own vendor admin panel.

Hi @timuric. Thank you for your answer. Can you share the website itself? Just so we get a feeling of how it works, what they could do, etc. I'm asking just for their web portal to see it as a client/visitor.

Another question, can I use Stripe Connect to split the payments between my company and the vendors in the Marketplace automatically?

And one more: do we have an option of selling Subscriptions through Saleor?

Thank you.

**Reply by ashishkpaul** [reply]
> Is there any plan for Saleor to ever go "Multi-Vendor" natively or do we need to need to just build custom? I know I asked this about three years ago because it was on the roadmap and then it disappeared.

There is another project on github which going to support multi-vendor marketplace https://github.com/orgs/vendure-ecommerce/projects/1/views/4

**Reply by looped-monk** [reply]
HI @ashishkpaul Ashish. Thanks much for sharing this. Is multi-vendor already part of it?

**Reply by ashishkpaul** [reply]
Please check the road map. Most probably, it will be fully developed by Q1 of 2023

**Reply by looped-monk** [reply]
@ashishkpaul Sure. thanks for sharing !

**Reply by timuric** [reply]
> > Is there any plan for Saleor to ever go "Multi-Vendor" natively or do we need to need to just build custom? I know I asked this about three years ago because it was on the roadmap and then it disappeared.
> 
> 
> 
> There is another project on github which going to support multi-vendor marketplace https://github.com/orgs/vendure-ecommerce/projects/1/views/4

As you can see definitions of multi vendor feature is not very clear in that thread. Saleor already supports quite few topics raised by Vendure community, like multi warehouse, channel permissions, permission groups, multi tenant setup (cloud only)

**Reply by timuric** [reply]
> > @leandrobortolotto yes we have, but am not allowed to share since they are customers of Saleor cloud. What you are looking for is most likely a custom admin panel that would allow managing some of the configuration of the store. You can use https://github.com/saleor/saleor-dashboard as a reference for building your own vendor admin panel.
> 
> 
> 
> Hi @timuric. Thank you for your answer. Can you share the website itself? Just so we get a feeling of how it works, what they could do, etc. I'm asking just for their web portal to see it as a client/visitor.
> 
> 
> 
> Another question, can I use Stripe Connect to split the payments between my company and the vendors in the Marketplace automatically?
> 
> 
> 
> And one more: do we have an option of selling Subscriptions through Saleor?
> 
> 
> 
> Thank you.
> 
> 
> 
> 

This is social platform that allows people to sell products in a tiktok like feed https://cargo.co/

Regarding payments you can orchestrate it however you wish, most simple form would make payouts to vendors

**Reply by ashishkpaul** [reply]
Hello @timuric , is there any guide,  link to achieve multi tenant setup

**Reply by leandrobortolotto** [reply]
@timuric is it ok if I contact Cargo's team to ask about some technical details on how they use Saleor? Thank you.

**Reply by leandrobortolotto** [reply]
Hello @timuric and everyone.

I was surfing on https://saleor.io and I found this text **Find your Saleor integrations below. _Saleor Marketplace is coming soon._** on the link https://saleor.io/integrations/.

Do you have an idea of when the Marketplace Integration will be released?

Thank you.

**Reply by timuric** [reply]
Hi @leandrobortolotto marketplace aka Saleor Appstore is already live, you can find apps in the dashboard we are shipping new apps on weekly basis, they are available as saas and open source version https://github.com/saleor/apps

**Reply by vitalijalbu** [reply]
Hi guys, I'd need to create a marketplace similar to **vinted**, as frontend I'm using react. So basically a user can:?
- create update own products 
- buy from other users
- sell and view his orders as a vendor
Thanks to anyone that can help me with that. 🦾

**Reply by vitalijalbu** [reply]
@timuric hi, can you help me with above question?
